src/robot_new.cpp

```diff
@@ -0,0 +1,396 @@
+     */
+    Kp_dl = new double(100.0);
+    Kd_dl = new double(0.0);
+    for (int joint = 0; joint < kinematics.number_of_dofs; joint++) {
+        ROS_INFO("initializing controllers for joint %d %s", joint, kinematics.joint_names[joint].c_str());
+        // connect and register the cardsflow state interface
+        hardware_interface::CardsflowStateHandle state_handle(kinematics.joint_names[joint], joint, &q[joint], &qd[joint],
+                                                              &qdd[joint], &kinematics.L, &kinematics.M, &kinematics.CG
+
+        );
+        cardsflow_state_interface.registerHandle(state_handle);
+
+        // connect and register the cardsflow command interface
+        hardware_interface::CardsflowHandle pos_handle(cardsflow_state_interface.getHandle(kinematics.joint_names[joint]),
+                                                       &q_target[joint], &qd_target[joint], &kinematics.torques[joint], &ld[joint],
+                                                       Kp_dl, Kd_dl);
+        cardsflow_command_interface.registerHandle(pos_handle);
+    }
+    registerInterface(&cardsflow_command_interface);
+
+    ekf_ = new BFL::KinDynEKF(kinematics.number_of_joints);
+
+    /**
+     * Initialize at zero for simulation
+     */
+    if(simulated){
+        q.setZero();
+        qd.setZero();
+        qdd.setZero();
+
+        kinematics.setRobotState(q, qd);
+        kinematics.updateJacobians();
+    }
+
+    try {
+        last_visualization =
+                ros::Time::now() - ros::Duration(10); // triggers immediate visualization in first iteratiom
+    }
+    catch(std::runtime_error& ex) {
+        ROS_ERROR("Exception: [%s]", ex.what());
+    }
+
+    int k=0;
+    nh->getParam("endeffectors", kinematics.endeffectors);
+    kinematics.endeffector_dof_offset.push_back(0);
+    Ld.resize(kinematics.endeffectors.size());
+    for (string ef:kinematics.endeffectors) {
+        ROS_INFO_STREAM("configuring endeffector " << ef);
+        vector<string> ik_joints;
+        nh->getParam((ef + "/joints"), ik_joints);
+        if (ik_joints.empty()) {
+            ROS_WARN(
+                    "endeffector %s has no joints defined, check your endeffector.yaml or parameter server.  skipping...",
+                    ef.c_str());
+            continue;
+        }
+        kinematics.endeffector_index[ef] = k;
+        kinematics.endeffector_number_of_dofs.push_back(ik_joints.size());
+        if(k>0)
+            kinematics.endeffector_dof_offset.push_back(kinematics.endeffector_dof_offset[k-1]+kinematics.endeffector_number_of_dofs[k-1]);
+
+        Ld[k].resize(kinematics.number_of_cables);
+        Ld[k].setZero();
+        k++;
+    }
+
+    updateSubscribers();
+}
+
+void Robot::update(){
+
+    if (nh->hasParam("k_dt"))
+        nh->getParam("k_dt", k_dt);
+    if (nh->hasParam("Kp_dl"))
+        nh->getParam("Kp_dl", *Kp_dl);
+    if (nh->hasParam("Kd_dl"))
+        nh->getParam("Kd_dl", *Kd_dl);
+
```

Comment:
maybe add a flag that would enable reading this params, something like `debug` maybe? that it defined in the launch file

Reply:
too avoid reading these parameters all the time